src/lib/Wst/Offchain/Scripts.hs

```diff
@@ -100,32 +101,39 @@ programmableLogicGlobalScript target paramsPolId =
   let script = Scripts.tryCompile target $ mkProgrammableLogicGlobal # pdata (pconstant $ transPolicyId paramsPolId)
   in C.PlutusScriptSerialised $ serialiseScript script
 
-permissionedTransferScript :: ScriptTarget -> C.Hash C.PaymentKey -> C.PlutusScript C.PlutusScriptV3
-permissionedTransferScript target cred =
-  let script = Scripts.tryCompile target $ mkPermissionedTransfer # pdata (pconstant $ transPubKeyHash cred)
-  in C.PlutusScriptSerialised $ serialiseScript script
-
 freezeTransferScript :: ScriptTarget -> C.PaymentCredential -> C.PolicyId -> C.PlutusScript C.PlutusScriptV3
 freezeTransferScript target progLogicBaseSpndingCred blacklistPolicyId =
   let script = Scripts.tryCompile target $ mkFreezeAndSeizeTransfer # pdata (pconstant $ transCredential progLogicBaseSpndingCred) # pdata (pconstant $ transPolicyId blacklistPolicyId)
   in C.PlutusScriptSerialised $ serialiseScript script
 
+{-| 'C.PlutusScript C.PlutusScriptV3' that always succeeds. Can be used for minting, withdrawal, spending, etc.
+-}
+alwaysSucceedsScript :: ScriptTarget -> C.PlutusScript C.PlutusScriptV3
+alwaysSucceedsScript target =
+  C.PlutusScriptSerialised $ serialiseScript $ Scripts.tryCompile target palwaysSucceed
+
+-- All the scripts below are nonced permissioned validators
+
+permissionedMintingScript :: ScriptTarget -> C.Hash C.PaymentKey -> C.PlutusScript C.PlutusScriptV3
+permissionedMintingScript target cred =
+  let script = Scripts.tryCompile target $ mkPermissionedMinting # pforgetData (pdata (pconstant "permissioned minting" :: ClosedTerm PByteString)) # pdata (pconstant $ transPubKeyHash cred)
+  in C.PlutusScriptSerialised $ serialiseScript script
+
+permissionedSpendingScript :: ScriptTarget -> C.Hash C.PaymentKey -> C.PlutusScript C.PlutusScriptV3
+permissionedSpendingScript target cred =
+  let script = Scripts.tryCompile target $ mkPermissionedTransfer # pforgetData (pdata (pconstant "permissioned spending" :: ClosedTerm PByteString)) # pdata (pconstant $ transPubKeyHash cred)
+  in C.PlutusScriptSerialised $ serialiseScript script
+
 blacklistMintingScript :: ScriptTarget -> C.Hash C.PaymentKey -> C.PlutusScript C.PlutusScriptV3
 blacklistMintingScript target cred =
-  let script = Scripts.tryCompile target $ mkPermissionedMinting # pdata (pconstant $ transPubKeyHash cred)
+  let script = Scripts.tryCompile target $ mkPermissionedMinting # pforgetData (pdata (pconstant "blacklist minting" :: ClosedTerm PByteString)) # pdata (pconstant $ transPubKeyHash cred)
```

Comment:
What's the different between "permissioned" minting / spending and "blacklist" minting / spending? If it's just nominal, why do we need both?

Reply:
No difference really, just though I'd nonce them differently to avoid confusions as it took me a while to debug as there were spending scripts, minting policies and stake scripts that had the same hash.